tests/test_packages.py

```diff
@@ -81,6 +82,128 @@ def test_simple_setuptools(tmp_path, env, nvcc_version):
     assert extras == {"test": {"dask-cuda==24.4.*,>=0.0.0a0"}}
 
 
+# rapids-build-backend should support projects using setuptools whose setup.py
+# file has 'import' statements depending on some project(s) that need to be extracted
+# from dependencies.yaml at build time
+def test_setuptools_with_imports_in_setup_py_works(
+    tmp_path,
+    isolated_env,
+):
+    package_dir = tmp_path / "pkg"
+    os.makedirs(package_dir)
+
+    template_args = {
```

Comment:
Minor nit: This is my fault for creating this pattern (I assume it was me), but on second read it's certainly a bit strange to use the same template args for all the `generate_from_template` calls. It would probably be cleaner to have just the necessary ones for each file.

Reply:
Yeah as far as I can tell, there's 0 sharing of template arguments across the files, so it should be possible to instead do something like:

```python
template_values = {
   "pyproject.toml": {
       "name": "something",
       "build_backend": "setuptools.build_meta
   },
   "dependencies.yaml": {
       "dependencies": {
            "cu11": ["cupy-cuda11x>=12.0.0"],
            "cu12": ["cupy-cuda12x>=12.0.0"],
        }
   }
}
generate_from_template(package_dir, template_args)
```

Where the top-level keys are the files to use and the dictionary passed after each one is the template arguments.

However... I'd rather not do that refactoring as part of this PR. I don't think that's THAT much better than the current state, at least with the small number of tests we currently have here.